sound/soc/sof/core.c

```diff
@@ -391,8 +391,6 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 
 	return 0;
 
-comp_err:
-	snd_soc_unregister_component(sdev->dev);
```

Comment:
Is this really helpful for fixing #861 or changing it while you are at here? My understanding is that we can invoke unregister specifically for resource-managed component device.

Reply:
yup, just while i'm at it. Not relevant for #861. Only the first commit is needed to fix the issue

Reply:
I'm not sure this is a good idea. This is a .probe() "bottom half," it's often performed in a work context after the actual probing has completed and returned success. Usually when you use managed allocations, it's done within .probe(), so that a failure leads to the .probe() failing and then all such allocations are freed. In this case an allocation failure will not lead to .probe() failing, so, the resource will stay allocated. This isn't bad, it will be freed if the modules are unloaded, but it's also not useful to keep it in the error case. So, maybe better to keep that unregister.

Reply:
@lyakh i havent tried this yet but what happens if you leave the unregister here, the machine driver registration fails and then you try to unload sof device?

Reply:
@ranj063 I haven't tried either. Nothing bad should happen at the resource management level, you have already unregistered the resource, so, no second unregistering will be attempted by the resource management system. But I don't know whether the driver's .release() method would choke because it assumes that the registration has completed successfully.

Reply:
@lyakh i just tried it and nothing bad happened. I am OK to leave it here or remove it. removing it makes more sense to me but if you prefer leaving it here i'm OK with it too.